### ISSUE TITLE
Is anyone interested in forking and fixing bug in markdown-it-span?  

### ISSUE THREAD

**Issue by oshihirii** [markdown-it/markdown-it#508]
There are two issues at:

https://github.com/pnewell/markdown-it-span

relating to a bug where:

**Desired output:**  

    ::spanned:: => <span>spanned</span>

    spanned: => spanned:

**Actual output:**  

    ::spanned:: => <span><span>spanned</span></span>

    spanned: => spanned:: 

Related Issues:

https://github.com/pnewell/markdown-it-span/issues/1 - "Span tag is creating twice"
https://github.com/pnewell/markdown-it-span/issues/3 - "Problem with colons that are not part of the syntax"

Would anyone be interested in forking and fixing bug?  

I can't figure out how to fix the bug and don't know how to update the npm package either.  

I just thought I would ask, the issues were opened in June and September so it doesn't look like they will receive a response, thank you.

**Comment by puzrin** [comment]
Here are no free programmers :). At short glance, that package looks like clone of https://github.com/markdown-it/markdown-it-ins. Original has [no such issues](https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%2B%2Binserted%2B%2B%5Cn%5Cninserted%2B%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D). I'd suggest you to make clone yourself, more carefully.

**Comment by oshihirii** [comment]
Thanks for linking to the plugin it was based on.  

I attempted forking the original plugin and started looking into how to create an npm package, but was too overwhelmed to learn that at the moment.  

So I did the following steps and it seems to be working.

I include them below for reference.  

`01)`  Download this file:

https://github.com/markdown-it/markdown-it-ins/blob/master/index.js

as:

     markdown-it-span.js

-------

`02)`  Replace initial file comments with:

    /*! markdown-it-span 1.0.0 based on:  https://github.com//markdown-it/markdown-it-ins 2.0.0 @license MIT */

-------

`03`  Replace the one instance of:

    markdownitIns

with:

    markdownitSpan

-------

`04)`  Replace the two instances of `0x2B` with `0x3A`.  

-------

`05)`  Replace the two instances of:

    token.markup  = '++';

with:

    token.markup  = '::';

-------

`06)`  Replace the one instance of:

    ins_plugin

with:

    span_plugin

-------

`07)`  Replace the one instance of:

    token.type    = 'ins_open';
    token.tag     = 'ins';

with:

    token.type    = 'span_open';
    token.tag     = 'span';

-------

`08)`  Replace the one instance of:

    token.type    = 'ins_close';
    token.tag     = 'ins';

with:

    token.type    = 'span_close';
    token.tag     = 'span';

-------

`09)`  Replace the one instance of:

    state.tokens[endDelim.token - 1].content === '+'

with:

    state.tokens[endDelim.token - 1].content === ':'

-------

`10)`  Replace the one instance of:

    (j < state.tokens.length && state.tokens[j].type === 'ins_close')

with:

    (j < state.tokens.length && state.tokens[j].type === 'span_close')

-------

`11)`  Replace the one instance of:

    md.inline.ruler.before('emphasis', 'ins', tokenize);
    md.inline.ruler2.before('emphasis', 'ins', postProcess);

with:

    md.inline.ruler.before('emphasis', 'span', tokenize);
    md.inline.ruler2.before('emphasis', 'span', postProcess);

-------

`12)`  Include in your `main.js` file with something like:

```js
// define your options
var md = require('markdown-it')({
    html: false,
    xhtmlOut: false,
    breaks: false,
    langPrefix: 'language-',
    linkify: false,
    typographer: false,
    quotes: '“”‘’',
    highlight: function( /*str, lang*/ ) { return ''; }
});

// import any other plugins
var markdownItAttrs = require('markdown-it-attrs');

// import from your js source directory
var markdownItSpan = require('./markdown-it-span');

// tell md to use the plugins  
md.use(markdownItAttrs).use(markdownItSpan);
```
-------

**Results:**

    ::spanned:: => <span>spanned</span>
    
    spanned: => spanned:
    
    // when using the markdown-it-attrs plugin as well
    ::spanned::{.my_class} => <span class="my_class">spanned</span>

**Comment by Netsaver** [comment]
I tried to apply all passages (except for 02 and 03, not applicable for current markdown-it-ins release) but for me the result was the same.. at every occurrence, the `:` symbol is duplicated, not only inside the span but everywhere in the text.
Honestly, I'm using tiddlywiki, but I've installed and tested successfully all other markdown-it plugins (attrs, mark, container, footnote... ).
I've noted that duplication is avoidable via backslash escaping: `\: --> :` (while  `:-->:: `) but for me backslash escaping gets the source messy and not reusable.
regards

**Comment by oshihirii** [comment]
@Netsaver  i’m not at pc now, but I’m pretty sure below is the code I used, I can confirm when back at pc, maybe it can help: 

```js
/*! markdown-it-span 1.0.0 based on:  https://github.com//markdown-it/markdown-it-ins 2.0.0 @license MIT */(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.markdownitSpan = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
'use strict';


module.exports = function span_plugin(md) {
  // Insert each marker as a separate text token, and add it to delimiter list
  //
  function tokenize(state, silent) {
    var i, scanned, token, len, ch,
        start = state.pos,
        marker = state.src.charCodeAt(start);

    if (silent) { return false; }

    if (marker !== 0x3A/* + */) { return false; }

    scanned = state.scanDelims(state.pos, true);
    len = scanned.length;
    ch = String.fromCharCode(marker);

    if (len < 2) { return false; }

    if (len % 2) {
      token         = state.push('text', '', 0);
      token.content = ch;
      len--;
    }

    for (i = 0; i < len; i += 2) {
      token         = state.push('text', '', 0);
      token.content = ch + ch;

      state.delimiters.push({
        marker: marker,
        jump:   i,
        token:  state.tokens.length - 1,
        level:  state.level,
        end:    -1,
        open:   scanned.can_open,
        close:  scanned.can_close
      });
    }

    state.pos += scanned.length;

    return true;
  }


  // Walk through delimiter list and replace text tokens with tags
  //
  function postProcess(state) {
    var i, j,
        startDelim,
        endDelim,
        token,
        loneMarkers = [],
        delimiters = state.delimiters,
        max = state.delimiters.length;

    for (i = 0; i < max; i++) {
      startDelim = delimiters[i];

      if (startDelim.marker !== 0x3A/* + */) {
        continue;
      }

      if (startDelim.end === -1) {
        continue;
      }

      endDelim = delimiters[startDelim.end];

      token         = state.tokens[startDelim.token];
      token.type    = 'span_open';
      token.tag     = 'span';
      token.nesting = 1;
      token.markup  = '::';
      token.content = '';

      token         = state.tokens[endDelim.token];
      token.type    = 'span_close';
      token.tag     = 'span';
      token.nesting = -1;
      token.markup  = '::';
      token.content = '';

      if (state.tokens[endDelim.token - 1].type === 'text' &&
          state.tokens[endDelim.token - 1].content === ':') {

        loneMarkers.push(endDelim.token - 1);
      }
    }

    // If a marker sequence has an odd number of characters, it's splitted
    // like this: `~~~~~` -> `~` + `~~` + `~~`, leaving one marker at the
    // start of the sequence.
    //
    // So, we have to move all those markers after subsequent s_close tags.
    //
    while (loneMarkers.length) {
      i = loneMarkers.pop();
      j = i + 1;

      while (j < state.tokens.length && state.tokens[j].type === 'span_close') {
        j++;
      }

      j--;

      if (i !== j) {
        token = state.tokens[j];
        state.tokens[j] = state.tokens[i];
        state.tokens[i] = token;
      }
    }
  }

  md.inline.ruler.before('emphasis', 'span', tokenize);
  md.inline.ruler2.before('emphasis', 'span', postProcess);
};

},{}]},{},[1])(1)
});
```

**Comment by Netsaver** [comment]
OK, I got the point, your code was right (and was the same I added as plugin).
It was my fault, because I updated the _markdown-it-span_ plugin with your code (with Tiddlywiki _repackage_ command) without uninstalling the previous release (original span plugin).
For some reason, also after the update, Tiddlywiki system kept both releases but in fact continued applying only the 1st one. Instead uninstalling and reinstalling the new one all worked perfectly.
So at the end I can confirm that both the _ins plugin_ and _ins refactored as span_ work as expected.
Thanks a lot,
Netsaver, Rome
